src/main/webapp/app/core/course/manage/quick-actions/quick-actions.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, inject, input } from '@angular/core';
+import { Router, RouterLink } from '@angular/router';
+import { ButtonComponent, ButtonSize, ButtonType } from 'app/shared/components/button/button.component';
+import { UserManagementDropdownComponent } from 'app/core/course/manage/user-management-dropdown/user-management-dropdown.component';
+import { Course } from 'app/core/course/shared/entities/course.model';
+import { faChalkboardUser, faChartBar, faClipboard, faGraduationCap, faListAlt, faQuestion } from '@fortawesome/free-solid-svg-icons';
+import { AddExerciseModalComponent } from 'app/core/course/manage/quick-actions/add-exercise-modal/add-exercise-modal.component';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { TranslateDirective } from 'app/shared/language/translate.directive';
+import { FeatureToggle } from 'app/shared/feature-toggle/feature-toggle.service';
+import { FaIconComponent } from '@fortawesome/angular-fontawesome';
+
+@Component({
+    selector: 'jhi-quick-actions',
+    templateUrl: './quick-actions.component.html',
+    styleUrls: ['./quick-actions.component.scss'],
+    imports: [ButtonComponent, UserManagementDropdownComponent, TranslateDirective, FaIconComponent, RouterLink],
+})
+export class QuickActionsComponent {
+    course = input<Course | undefined>();
+    protected readonly ButtonType = ButtonType;
+    protected readonly ButtonSize = ButtonSize;
+    protected readonly faListAlt = faListAlt;
+    protected readonly faChartBar = faChartBar;
+    protected readonly faClipboard = faClipboard;
+    protected readonly faGraduationCap = faGraduationCap;
+    protected readonly faChalkboardUser = faChalkboardUser;
+    protected readonly faQuestion = faQuestion;
+    private router = inject(Router);
+    private modalService = inject(NgbModal);
+
+    linkToLectureCreation() {
+        if (!this.course()?.id) {
+            return;
+        }
+        this.router.navigate(['/course-management', this.course()?.id, 'lectures', 'new']);
+    }
+    linkToExamCreation() {
+        if (!this.course()?.id) {
+            return;
+        }
+        this.router.navigate(['/course-management', this.course()?.id, 'exams', 'new']);
+    }
+    linkToFaqCreation() {
+        if (!this.course()?.id) {
+            return;
+        }
+        this.router.navigate(['/course-management', this.course()?.id, 'faqs', 'new']);
+    }
```

Comment:
This seems quite redundant, I think it would be better to have a `getCourseManagementLink` method and define an enum with the values `lectures`, `exams`, `faqs` of which a value is expected as parameter